src/mmap.rs

```diff
@@ -60,6 +65,8 @@ pub enum MmapError {
 pub struct GuestRegionMmap {
     mapping: MmapRegion,
     guest_base: GuestAddress,
+    file: Option<File>,
```

Comment:
@sboeuf Can you investigate `MmapRegion` instead? It sounds logical. I think if you move it it would also be possible to switch to the @alexandruag's proposed single `Option<File, usize>`

Reply:
I think this is feasible, but do you expect me to implement this both for `mmap_unix.rs` and `mmap_windows.rs`?

I don't really like to merge the file and the file offset into a tuple, because it's less readable from a code perspective IMO.

Reply:
Just for `mmap_unix.rs`, since in its current state this source file is Unix-specific anyway :P We can reach out to the Windows folks regarding also updating the `mmap_windows.rs` file.

Reply:
Ah yes you got a point with the `unwrap` here!

Reply:
I remember one of the reason I didn't create `File` as part of the `MmapRegion`. I wanted the region to own the File. And to do so, I had to invoke `try_clone()`, which should not be done at the same level the `mmap()` is done IMO.
Both unix and windows should share the same ownership behavior regarding the File I think.

Reply:
@jiangliu what do you think about this?

Reply:
It's feasible to move (file, offset) pair into MmapRegion. I'm also thinking of adding an method to the GuestMemoryRegion trait to expose the (file, offset) info. Then clients like vhost may depend on GuestMemoryRegion instead of GuestRegionMmap.